### DISCUSSION TITLE
Call FGPropeller::Calculate in JSBSim interface

### DISCUSSION THREAD

**Discussion by NOTaShinigami** [JSBSim-Team/jsbsim#783] [Q&A]
Hello, I'm working on a project that requires the use of MATLAB and JSBSim. As part of my project, I want to adjust the RPM of my propeller at a particular power. I discovered in the FGPropeller class a Calculate member function that performs the work for me and that I'd like to utilize in the SFunction interface code. Is it possible to access it or is there some other solution to the problem?

**Comment by bcoconni** [comment]
Hi @NOTaShinigami,
At first, I'd say that the C++ code of the S-Function would need some adjustments to make some direct calls to `FGPropeller::Calculate()`. @noah-curran, do you concur ?
However, needing to call `FGPropeller::Calculate()` is quite unusual: in most cases the propeller RPM computation is left to JSBSim. Could you be more specific about the reasons you'd want to call `FGPropeller::Calculate()` ?

**Reply by noah-curran** [reply]
It's definitely possible to make custom adjustments to the S-Function for this, but I have a hard time seeing it being done smoothly without impacting other projects that may use the S-Function, mainly due to your observation that this sort of thing would be left to JSBSim.

**Reply by NOTaShinigami** [reply]
The reason I want to call 'FGPropeller::Calculate()' is that I have a controller created in Matlab that outputs the power required to maintain a manoeuvre, and I'd like to control the RPM of my propellers, and consequently the throttle command, using this power. I saw the calculate function and thought it would be a good way to accomplish this integration, but if you believe there are other techniques, I am open to thoughts and input.

**Reply by agodemar** [reply]
@NOTaShinigami please provide more infos. Dou you have a variable pitch propeller? Which type of aircraft? Are you able to model a PID controller in Simulink and adjust the blade pitch angle as @dpculp suggests below?

**Comment by dpculp** [comment]
In JSBSim you can adjust the propeller RPM by adjusting the blade pitch angle while maintaining a constant throttle.  To gain manual control of the blade angle you first set "constant-speed-mode" to zero.  Here's an image of a test I just now ran in FlightGear.  On the left are some engine and propeller parameters when the blade angle is set to 15 degrees, and on the right set to 20 degrees.

![prop-test](https://user-images.githubusercontent.com/21369577/209589374-6c0f88b5-373e-43b8-8302-82b831cefb31.jpg)

The engine power output changes even though the throttle setting is constant, so if you really need a constant engine power I recommend using an electric engine instead.
